django/test/utils.py

```diff
@@ -127,6 +145,11 @@ def teardown_test_environment():
     Template._render = Template._original_render
     del Template._original_render
 
+    for engine in engines:
+        engine_cls = engines[engine]
+        if hasattr(engine_cls, '_original_render'):
+            engine_cls.render = engine_cls._original_render
```

Comment:
These checks should be run against `engine_cls.template_class` and not `engine_cls`.